.github/workflows/release.yaml

```diff
@@ -0,0 +1,280 @@
+      matrix:
+        os: [linux, darwin, windows]
+        arch: [amd64, arm64]
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.extract-version-details.outputs.full-version }}
+      - uses: cachix/install-nix-action@v18
+        with:
+          extra_nix_config: |
+            keep-derivations = true
+            keep-outputs = true
+      - run: |
+          GOOS=${{ matrix.os }} \
+            GOARCH=${{ matrix.arch }} \
+            CGO_ENABLED=0 \
+            LDFLAGS="-w -s -X github.com/rwx-research/captain-cli.Version=${{ needs.extract-version-details.outputs.full-version }}" \
+            nix develop --command mage
+      - if: ${{ matrix.os != 'windows' }}
+        run: |
+          if [[ "${{ matrix.arch }}" == "amd64" ]]; then
+            new_arch="x86_64"
+          elif [[ "${{ matrix.arch }}" == "arm64" ]]; then
+            new_arch="aarch64"
+          else
+            echo "unmapped arch: ${{ matrix.arch }}"
+            exit 1
+          fi
+
+          github_asset_name=$(echo "captain-${{ matrix.os }}-$new_arch" | tr '[:upper:]' '[:lower:]')
+          mv captain "$github_asset_name"
+          gh release upload ${{ needs.extract-version-details.outputs.full-version }} "$github_asset_name" --clobber
+      - if: ${{ matrix.os == 'windows' }}
+        run: |
+          if [[ "${{ matrix.arch }}" == "amd64" ]]; then
+            new_arch="x86_64"
+          elif [[ "${{ matrix.arch }}" == "arm64" ]]; then
+            new_arch="aarch64"
+          else
+            echo "unmapped arch: ${{ matrix.arch }}"
+            exit 1
+          fi
+
+          github_asset_name=$(echo "captain-${{ matrix.os }}-$new_arch.exe" | tr '[:upper:]' '[:lower:]')
+          mv captain.exe "$github_asset_name"
+          gh release upload ${{ needs.extract-version-details.outputs.full-version }} "$github_asset_name" --clobber
+
+  obtain-approval:
+    runs-on: ubuntu-latest
+    if: ${{ inputs.kind == 'production' }}
+    needs:
+      - build-and-upload-binaries
+    environment: release-approval
+    steps:
+      - run: exit 0
+
+  publish-release:
+    runs-on: ubuntu-latest
+    if: ${{ always() && (needs.obtain-approval.result == 'success' || needs.obtain-approval.result == 'skipped') }}
+    needs:
+      - extract-version-details
+      - obtain-approval
+    permissions:
+      contents: write
+      id-token: write
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.extract-version-details.outputs.full-version }}
+      - name: Publish the full version release
+        run: |
+          gh release edit ${{ needs.extract-version-details.outputs.full-version }} \
+            --draft=false \
+            --latest=${{ inputs.kind == 'production' }}
+      - name: Push the aliased tags
+        if: ${{ needs.extract-version-details.outputs.aliased-version != '' }}
```

Comment:
might warrant a comment here about this being empty for testing releases or asserting on `inputs.kind` to clarify why this conditional is there

Reply:
Is it not clear as-is? 🤔 If there is no aliased-version set, don't push aliased tags and make an aliased release?

I didn't tie this to the kind because it's not so much related to the kind as it is the aliased version (e.g. it doesn't care that for production, it's v1 or unstable it's unstable)

Reply:
when I was reading I asked, "when is the there no aliased tag", "is this defensive code?". until I looked back and realized the it's empty for `testing` releases.

(none of the feedback in this PR should block merge though!)